src/libsystemd/sd-journal/journal-file.c

```diff
@@ -543,6 +574,9 @@ static int journal_file_verify_header(JournalFile *f) {
 
                 if (f->header->field_hash_table_size == 0 || f->header->data_hash_table_size == 0)
                         return -EBADMSG;
+
+                if (JOURNAL_FILE_COMPRESSION(f) != compression_requested())
+                        return -EPROTONOSUPPORT;
```

Comment:
If a file is using a given compression algorithm, and we support it, why not just continue using the file with that algorithm?

Reply:
OK. But in that case, we need to explicitly rotate journal to enable the specified algorithm.

Reply:
please return a recognizable error and then treat this as a reason for rotation further up the stack